Editor/Scripts/Analytics/MeshSyncObserver.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+
+using UnityEditor;
+
+namespace Unity.MeshSync.Editor.Analytics {
+
+    /// <summary>
+    /// Hacky way to restore observers to MeshSync servers in scene
+    /// </summary>
+    [InitializeOnLoad]
+    internal static class MeshSyncObserverStartUp {
+        private const string SESSION_STATE_FLAG = "meshSyncAnalyticsSetupFlag";
+        private static DateTime m_check_Start;
+        private static readonly TimeSpan INTERVAL = TimeSpan.FromSeconds(3);
+
+        private static void updateCallback(bool limitChecks) {
+            // Limit checks to once every 3 seconds
+            if ((DateTime.Now - m_check_Start < INTERVAL) && limitChecks) {
+                return;
+            }
+
+            m_check_Start = DateTime.Now;
+
+            var alreadySetup = SessionState.GetBool(SESSION_STATE_FLAG, defaultValue: false);
+
+            // If the setup was already done, we still need to do it again at least once after a domain reload
+            if (alreadySetup) {
+                EditorApplication.update -= checkWithTimeLimits;
+            }
+
+            var array = UnityEngine.Object.FindObjectsOfType<BaseMeshSync>(includeInactive: true);
+            bool success = false;
+
+            foreach (var server in array) {
+                if (server.getNumObservers == 0) {
+                    var observer = new MeshSyncObserver();
+                    server.Subscribe(observer);
+                    success = true;
+                }
+            }
+
+            if (success) {
+                SessionState.SetBool(SESSION_STATE_FLAG, value: true);
+                EditorApplication.update -= checkWithTimeLimits;
+            }
+        }
+
+        static MeshSyncObserverStartUp() {
+            m_check_Start = DateTime.Now;
+
+            EditorApplication.update -= checkWithTimeLimits;
+            EditorApplication.update += checkWithTimeLimits;
+
+            EditorApplication.playModeStateChanged -= playModeChangedCallback;
+            EditorApplication.playModeStateChanged += playModeChangedCallback;
+
+            EditorApplication.hierarchyChanged -= checkWithoutTimeLimits;
+            EditorApplication.hierarchyChanged += checkWithoutTimeLimits;
+            Undo.undoRedoPerformed -= checkWithoutTimeLimits;
```

Comment:
Do we need to register the same callback for `Undo.undoRedoPerformed` in addition to `EditorApplication.hierarchyChanged` ?

Undoing an MeshSyncServer creation will already trigger `EditorApplication.hierarchyChanged`, so I think we only need the `EditorApplication.hierarchyChanged` one.

Reply:
I tested this and I confirm this is correct. The issue must have been there with undo because we were registering checkWithTimeLimits, so the event was missed because of the interval check.

Reply:
I removed the Undo callback subscription as suggested